tests/Microsoft.Identity.Test.Unit/CacheTests/TokenCacheNotificationTests.cs

```diff
@@ -269,5 +270,38 @@ await app
                 Assert.IsFalse(cacheAccessRecorder4.LastAfterAccessNotificationArgs.HasTokens);
             }
         }
+
+        [TestMethod]
+        public async Task TestNotSubscribedAsync()
```

Comment:
@bgavrilMS I was thinking about a unit test something like this. Mock `AppTokenCacheInternal` and check if cache events are called. Adding serialization cache (L287-288) should break this test, but currently it does not. Mu guess is we need to add implementation to the mocked token cache.

Does this approach seem reasonable?

Reply:
Yes, it need a bit more mocking power, but it works now.